source/features/quick-fork-deletion.tsx

```diff
@@ -0,0 +1,68 @@
+import './wait-for-build.css';
+import delay from 'delay';
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import {getRepo} from '../github-helpers';
+import pluralize from '../helpers/pluralize';
+import * as api from '../github-helpers/api';
+import elementReady from 'element-ready';
+
+async function handleCancelClick(event: delegate.Event): Promise<void> {
+	const button = event.delegateTarget;
+	event.preventDefault();
+	button.classList.add('btn-sm');
+	button.textContent = 'Delete fork';
+}
+
+async function handleFirstClick(event: delegate.Event): Promise<void> {
+	const button = event.delegateTarget;
+	button.classList.remove('btn-sm');
+	event.preventDefault();
+
+	let secondsLeft = 5;
+	do {
+		button.textContent = `Deleting fork in ${pluralize(secondsLeft, '$$ second')}`;
+		await delay(1000); // eslint-disable-line no-await-in-loop
+	} while (--secondsLeft);
```

Comment:
❤️

Reply:
Not sure about all of this, I just wanted to create a PR 🤣

Reply:
I hate this pattern, but can't think of anything better (recursion would be nice I guess...)

Reply:
Yikes, no:

```js
	return (async function down(secondsLeft: number): Promise<boolean> {
		if (abortController.signal.aborted) {
			return false;
		}

		if (!secondsLeft) {
			return true;
		}

		button.textContent = `Deleting fork in ${pluralize(secondsLeft, '$$ second')}. Cancel?`;
		await delay(1000);
		return down(secondsLeft - 1);
	})(5);
```